extensions/resteasy-reactive/rest/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java

```diff
@@ -1049,21 +1049,21 @@ private boolean isRecord(IndexView index, DotName name) {
 
     private AnnotationInstance createTypedAnnotationInstance(ClassInfo clazz,
             BeanArchiveIndexBuildItem beanArchiveIndexBuildItem) {
-        Set<DotName> interfaceNames = new HashSet<>();
+        Set<DotName> allBeanTypes = new HashSet<>();
+        allBeanTypes.add(clazz.name());
+
         ClassInfo currentClazz = clazz;
-        while (!ResteasyReactiveDotNames.OBJECT.equals(currentClazz.name())) {
-            currentClazz.interfaceNames().forEach(iface -> interfaceNames.add(iface));
-            // inspect super class
+        while (!ResteasyReactiveDotNames.OBJECT.equals(currentClazz.name()) && currentClazz != null) {
+            if (currentClazz.isAbstract()) {
```

Comment:
Maybe all classes, even non abstract need to be added to the typed annotation?

Eg:
```
@Named("base")
public class Base {

}

@ApplicationScoped
public class Ext1 extends Base {
        @Context
        HttpHeaders headers;
}

@ApplicationScoped
public class Ext2 extends Base {
        @Context
        HttpHeaders headers;
}
```

Now doing an:
```
@Inject
Instance<Base>
```

what happens? I believe Instancehandle should point to Ext1 and Ext2, but won't since the Typed annotation does not contain the Base class.

Reply:
That's what I did first and I had ~ 4 tests failing with ambiguous resolution so I wonder if it might break some applications.
For instance `SimpleQuarkusRestTestCase` was failing and there were a few others.

Now I could be convinced to do it if it has a lot of value but we would have to check with @geoand .

Reply:
I personally would not do it for now

Reply:
ok, fine by me, at least this decission is now documented somewhere.